src/System.Reflection.TypeExtensions/tests/EventInfoTests.cs

```diff
@@ -0,0 +1,184 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+#pragma warning disable 0067
+
+namespace System.Reflection.Tests
+{
+    public delegate void VoidDelegate();
+    public delegate void IntDelegate(int i);
+
+    public class EventInfoTests
+    {
+        public static IEnumerable<object[]> AddEventHandler_TestData()
+        {
+            EI_Class tc1 = new EI_Class();
+            yield return new object[] { typeof(EI_Class).GetEvent("PublicEvent"), tc1, new VoidDelegate(tc1.PublicVoidMethod1), 1 };
+            yield return new object[] { typeof(EI_Class).GetEvent("PublicStaticEvent"), null, new VoidDelegate(tc1.ProtectedInternalVoidMethod), 2 };
+            yield return new object[] { typeof(EI_Class).GetEvent("PublicStaticEvent"), tc1, new VoidDelegate(tc1.PublicVoidMethod2), 3 };
+        }
+
+        [Theory]
+        [MemberData(nameof(AddEventHandler_TestData))]
+        public void AddEventHandler_RemoveEventHandler(EventInfo eventInfo, EI_Class target, Delegate handler, int expectedStaticVariable)
+        {
+            // Add and make sure we bound the event.
+            eventInfo.AddEventHandler(target, handler);
+            (target ?? new EI_Class()).InvokeAllEvents();
```

Comment:
It's not a big deal, but it seems like it'd be better to have two methods, InvokeInstanceEvents and InvokeStaticEvents, and then do:
```C#
target?.InvokeInstanceEvents();
EI_Class.InvokeStaticEvents();
```